frame/beefy/src/lib.rs

```diff
@@ -299,6 +299,30 @@ pub mod pallet {
 			Self::validate_unsigned(source, call)
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+			// weight used by `on_finalize()` hook
+			frame_support::weights::constants::RocksDbWeight::get().reads(1)
+		}
+
+		fn on_finalize(n: BlockNumberFor<T>) {
```

Comment:
If I understand correctly this is needed because `genesis_block` can be a future block. But do we need to allow `genesis_block` to be a future block ? Could we just reset from the current block ?